myenumerables.rb

```diff
@@ -0,0 +1,114 @@
+module Enumerable
+  def my_each
+    index = 0
+    while index < size
+      return to_a unless block_given?
+
+      yield self[index]
+      index += 1
+      self
+    end
+  end
+
+  def my_each_with_index(index = 0)
+    while index < size
+      return self unless block_given?
+
+      yield self[index], index
+      index += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each { |item| result << item if yield(item) == true }
+    result
+  end
+
+  def my_all?(query = nil)
+    my_each do |idx|
+      if query.is_a? Class
+        return false unless idx.is_a? query
+      elsif query.is_a? String or query.is_a? Integer
+        return false unless idx == query
+      elsif query.is_a? Regexp
+        return false unless idx.match(query)
+      elsif block_given?
+        return false unless yield(idx)
+      elsif !empty?
+        return false
+      end
+    end
+    true
+  end
+
+  def my_any?(*query)
+    if !query.empty?
+      my_each { |item| return true if item == query }
+    elsif block_given?
+      my_each { |item| return true if yield(item) }
+    else
+      my_each { |item| return true if item }
+    end
+    false
+  end
+
+  def my_none?(query = nil)
+    if !query.nil?
+      to_a.my_each { |item| return false if query == item }
+    elsif query.is_a? Class
+      my_each { |item| return false if item == query }
+    elsif block_given?
+      my_each { |item| return false if yield(item) }
+    else
+      my_each { |item| return false if item }
+    end
+    true
+  end
+
+  def my_count(query = nil)
+    count = []
```

Comment:
* [ ] There should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- ```p ary.my_count{ |x| x%2==0 }```should be equal to```p ary.count{ |x| x%2==0 } ```